src/goto-programs/interpreter.cpp

```diff
@@ -69,12 +69,12 @@ void interpretert::initialize(bool init)
     main_it=goto_functions.function_map.find(goto_functionst::entry_point());
 
   if(main_it==goto_functions.function_map.end())
-    throw "main not found";
+    throw analysis_exceptiont("main not found");
```

Comment:
Is this actually a structural invariant on goto-programs?

Reply:
I wasn't sure about this one, because you might have goto programs without an entry point (libraries mostly). I suppose it could be a precondition if we assume that a caller should've checked for an entry point before calling this.

Reply:
It is getting in to the heart of why we need to check that goto-programs are well formed and to clarify what "normal forms" exist for them.  If it has been linked and is a complete goto program then it should have an entry point which is a function in the map.  If it is only partially built (goto-cc -c for example) then it may not do.  It is not immediately clear to me whose fault it would be to try to interpret a n "object file" rather than an "executable".